### ISSUE TITLE
Change email address workflow is broken, results in empty JSON string being shown in browser

### ISSUE THREAD

**Issue by KathyReid** [common-voice/common-voice#3059]
**Describe the bug**
If an authenticated Common Voice user tries to change their email address from the page at
https://commonvoice.mozilla.org/en/profile/settings
they are redirected via SSO to 
https://auth.mozilla.auth0.com/login?
This interface is intuitive, and they need to enter the _new_ email address in the login box. 
related: #2110 for UI awkwardness

If they persist, and enter a _new_ email address, a confirmation email is sent to the _new_ email address. 
Clicking on the confirmation email results in the following being shown in the browser window: 

```
{

    "message": ""

}
```

**Expected behavior**
Easily able to change email address of account

**Screenshots**

**Desktop or Mobile (please complete the following information):**
- first reported on Windows 10 64 bit Firefox desktop
- validated by Kathy Reid on Linux Ubuntu 20.04 LT on Firefox 87

 **Additional Hardware (were you using headphones, an external speaker or an external microphone?):**
Not relevant to use case

**Comment by phirework** [comment]
Are you able to log in with the new email address after this first flow or is the account still under the old address?

**Comment by KathyReid** [comment]
Ah, good question, I should have tested that. 
No, clicking the link in the email that is generated lands the user back to the login screen where upon entering the email address, it generates another email with a link. And so on. 

I shall call this a "Kafka-esque design pattern" :D

**Comment by phirework** [comment]
It is honestly rude that GH doesn't allow 😭 as a reaction. Will investigate, thanks!

**Comment by phirework** [comment]
Hmm, I just tried changing my email and was able to successfully do so, can you verify if this still persists? I've tested email to email, email to Google login, Google login to email, Github to email. 

The only time I was able to get an empty JSON string is when I tried to change a regular email to a Google login attached to my Mozilla LDAP, I got an error message asking me to use password-based login instead, so I hit back, entered my Mozilla LDAP, and THEN it failed. But that seems like a one-use token issue rather than a login issue.

**Comment by KathyReid** [comment]
Firstly apologies for taking so long to come back around on this. 

* From https://commonvoice.mozilla.org/en/profile/settings, I clicked 'Edit' next to the email box 
* This brought up the SSO screen, and I typed in the new email address
* An email was generated to my new email address
* I opened that in a new browser session, and this was now set as my CV login email 

I repeated the process to be sure, and changed my new email address back to my old email address. 
My old email address is also linked to my GitHub account, and I also tested logging in with this, and it worked correctly. 

Should I go back to the original reporter to re-test?

**Comment by phirework** [comment]
Yeah, let's try to have them re-test. The auth system is largely out of my control unfortunately as it's integrated with Mozilla IAM, and I haven't made any changes to the client code in a while, so it could've been a big on the SSO side that has since been fixed.

**Comment by KathyReid** [comment]
No worries, I will go back to the issue reporter, and come back to you on this one. 
Feel free to assign this Issue to me in the meantime as the action is with me to follow up.

**Comment by moz-bozden** [comment]
Hi @KathyReid, in the last months a couple of users also had problems with changing e-mail addresses, but they could finally accomplish this.
Do you think this is still a problem?
